### ISSUE TITLE
Error playing Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv

### ISSUE THREAD

**Issue by ckrzen** [keredson/gnomecast#99]
Video works! No sound!

Confirmed that file _does_ play sound in mpv(mplayer).

------------------------------------------------------------

File: Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv
Video: Video #1 (h264)
Audio: eng (opus,)
Subtitles: 
Device: Chromecast (Unknown manufacturer)
Chromecast: v1.9.3

FileMetadata(fn:/home/chris/Downloads/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv, ready:True, thumbnail_fn:/tmp/gnomecast_thumbnail_74i4v5hd.jpg, container:mkv, video_streams:[StreamMetadata(index:0:0, codec:h264, title:Video #1)], audio_streams:[StreamMetadata(index:0:1, codec:opus,, title:eng, mono:False)], subtitles:[])

```ffmpeg version 4.1.4 Copyright (c) 2000-2019 the FFmpeg developers
Stream #0:0: Video: h264 (High), yuv420p(tv, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 29.97 fps, 29.97 tbr, 1k tbn, 59.94 tbc (default)
Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)```

**Comment by ckrzen** [comment]
DEBUG:


$ gnomecast 
update_button_visible
None
update_button_visible
update_button_visible
serving on http://192.168.1.72:38363
check_ffmpeg
b'/usr/bin/ffmpeg\n'
Chromecast('192.168.1.70', port=8009, device=DeviceStatus(friendly_name='Poodle Lounge TV', model_name='Chromecast', manufacturer='Unknown manufacturer', uuid=UUID('29466b4d-975e-ea46-7fdd-9c91dc866bff'), cast_type='cast'))
update_button_visible
update_button_visible
update_button_visible
Open clicked
File selected: ['/home/chris/Downloads/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv']
update_button_visible
STOP command requested but no session is active.
STOP command requested but no session is active.
update_button_visible
update_button_visible
update_button_visible
video: Stream #0:0: Video: h264 (High), yuv420p(tv, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 29.97 fps, 29.97 tbr, 1k tbn, 59.94 tbc (default) '0:0'
audio: Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
FileMetadata(fn:/home/chris/Downloads/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv, ready:True, thumbnail_fn:/tmp/gnomecast_thumbnail_a276a1_2.jpg, container:mkv, video_streams:[StreamMetadata(index:0:0, codec:h264, title:Video #1)], audio_streams:[StreamMetadata(index:0:1, codec:opus,, title:eng, mono:False)], subtitles:[])
Transcoder /home/chris/Downloads/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv
transcode, transcode_video, transcode_audio True False True
['ffmpeg', '-i', '/home/chris/Downloads/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv', '-map', '0:0', '-map', '0:1', '-c:v', 'copy', '-c:a', 'ac3', '-b:a', '256k', '/var/tmp/gnomecast_fw41s6la.mp4']
Video #1 - eng StreamMetadata(index:0:0, codec:h264, title:Video #1) StreamMetadata(index:0:1, codec:opus,, title:eng, mono:False)
update_button_visible
update_button_visible
chose subtitle No subtitles. None None
{'--prefix': '/usr', '--bindir': '/usr/bin', '--datadir': '/usr/share/ffmpeg', '--docdir': '/usr/share/doc/ffmpeg', '--incdir': '/usr/include/ffmpeg', '--libdir': '/usr/lib64', '--mandir': '/usr/share/man', '--arch': 'x86_64', '--optflags': "'-O2", '-Werror': 'format-security', '-Wp,-D_FORTIFY_SOURCE': '2', '-specs': '/usr/lib/rpm/redhat/redhat-hardened-ld', '-mtune': 'generic', '--extra-ldflags': "'-Wl,-z,relro", '--extra-cflags': "'", '--shlibdir': '/usr/lib64', 'q': '-1.0', 'frame': '1528', 'fps': '0.0', 'size': '7936kB', 'time': '00:00:50.93', 'bitrate': '1276.3kbits/s', 'speed': '102x'}
{'frame': '3191', 'fps': '3191', 'q': '-1.0', 'size': '16640kB', 'time': '00:01:46.41', 'bitrate': '1281.0kbits/s', 'speed': '106x'}
{'frame': '4726', 'fps': '3150', 'q': '-1.0', 'size': '25344kB', 'time': '00:02:37.63', 'bitrate': '1317.1kbits/s', 'speed': '105x'}
{'frame': '6343', 'fps': '3171', 'q': '-1.0', 'size': '34304kB', 'time': '00:03:31.61', 'bitrate': '1328.0kbits/s', 'speed': '106x'}
{'frame': '7953', 'fps': '3181', 'q': '-1.0', 'size': '43264kB', 'time': '00:04:25.30', 'bitrate': '1335.9kbits/s', 'speed': '106x'}
{'frame': '9532', 'fps': '3177', 'q': '-1.0', 'size': '54528kB', 'time': '00:05:17.99', 'bitrate': '1404.7kbits/s', 'speed': '106x'}
{'frame': '11157', 'fps': '3187', 'q': '-1.0', 'size': '64768kB', 'time': '00:06:12.21', 'bitrate': '1425.5kbits/s', 'speed': '106x'}
{'frame': '12711', 'fps': '3177', 'q': '-1.0', 'size': '75008kB', 'time': '00:07:04.06', 'bitrate': '1449.0kbits/s', 'speed': '106x'}
{'frame': '14251', 'fps': '3166', 'q': '-1.0', 'size': '85760kB', 'time': '00:07:55.45', 'bitrate': '1477.6kbits/s', 'speed': '106x'}
{'frame': '15798', 'fps': '3159', 'q': '-1.0', 'size': '97536kB', 'time': '00:08:47.06', 'bitrate': '1516.0kbits/s', 'speed': '105x'}
{'frame': '17317', 'fps': '3148', 'q': '-1.0', 'size': '107520kB', 'time': '00:09:37.75', 'bitrate': '1524.5kbits/s', 'speed': '105x'}
{'frame': '18859', 'fps': '3143', 'q': '-1.0', 'size': '113408kB', 'time': '00:10:29.20', 'bitrate': '1476.5kbits/s', 'speed': '105x'}
{'frame': '20494', 'fps': '3153', 'q': '-1.0', 'size': '122368kB', 'time': '00:11:23.77', 'bitrate': '1466.0kbits/s', 'speed': '105x'}
{'frame': '22089', 'fps': '3155', 'q': '-1.0', 'size': '132096kB', 'time': '00:12:16.98', 'bitrate': '1468.3kbits/s', 'speed': '105x'}
{'frame': '23675', 'fps': '3156', 'q': '-1.0', 'size': '141312kB', 'time': '00:13:09.89', 'bitrate': '1465.5kbits/s', 'speed': '105x'}
{'frame': '25251', 'fps': '3156', 'q': '-1.0', 'size': '150016kB', 'time': '00:14:02.48', 'bitrate': '1458.7kbits/s', 'speed': '105x'}
{'frame': '26829', 'fps': '3156', 'q': '-1.0', 'size': '158208kB', 'time': '00:14:55.13', 'bitrate': '1447.9kbits/s', 'speed': '105x'}
{'frame': '28307', 'fps': '3145', 'q': '-1.0', 'size': '166400kB', 'time': '00:15:44.45', 'bitrate': '1443.3kbits/s', 'speed': '105x'}
{'frame': '29897', 'fps': '3147', 'q': '-1.0', 'size': '174336kB', 'time': '00:16:37.53', 'bitrate': '1431.7kbits/s', 'speed': '105x'}
{'frame': '31445', 'fps': '3144', 'q': '-1.0', 'size': '185344kB', 'time': '00:17:29.15', 'bitrate': '1447.2kbits/s', 'speed': '105x'}
{'frame': '33046', 'fps': '3147', 'q': '-1.0', 'size': '194816kB', 'time': '00:18:22.58', 'bitrate': '1447.4kbits/s', 'speed': '105x'}
{'frame': '33386', 'fps': '2786', 'q': '-1.0', 'size': '197120kB', 'time': '00:18:33.92', 'bitrate': '1449.7kbits/s', 'speed': '92.9x'}
{'frame': '34653', 'fps': '2776', 'q': '-1.0', 'size': '204544kB', 'time': '00:19:16.19', 'bitrate': '1449.3kbits/s', 'speed': '92.6x'}
{'frame': '36170', 'fps': '2786', 'q': '-1.0', 'size': '213504kB', 'time': '00:20:06.81', 'bitrate': '1449.3kbits/s', 'speed': '92.9x'}
{'frame': '37769', 'fps': '2801', 'q': '-1.0', 'size': '224256kB', 'time': '00:21:00.16', 'bitrate': '1457.8kbits/s', 'speed': '93.4x'}
{'frame': '39369', 'fps': '2815', 'q': '-1.0', 'size': '234752kB', 'time': '00:21:53.55', 'bitrate': '1464.0kbits/s', 'speed': '93.9x'}
{'frame': '40944', 'fps': '2827', 'q': '-1.0', 'size': '245760kB', 'time': '00:22:46.10', 'bitrate': '1473.7kbits/s', 'speed': '94.3x'}
{'frame': '41598', 'fps': '2727', 'q': '-1.0', 'size': '250624kB', 'time': '00:23:07.92', 'bitrate': '1479.3kbits/s', 'speed': '91x'}
{'frame': '43163', 'fps': '2739', 'q': '-1.0', 'size': '260608kB', 'time': '00:24:00.15', 'bitrate': '1482.4kbits/s', 'speed': '91.4x'}
{'frame': '44756', 'fps': '2753', 'q': '-1.0', 'size': '271104kB', 'time': '00:24:53.30', 'bitrate': '1487.2kbits/s', 'speed': '91.9x'}
{'frame': '46316', 'fps': '2764', 'q': '-1.0', 'size': '281088kB', 'time': '00:25:45.35', 'bitrate': '1490.1kbits/s', 'speed': '92.2x'}
{'frame': '47809', 'fps': '2770', 'q': '-1.0', 'size': '291584kB', 'time': '00:26:35.19', 'bitrate': '1497.4kbits/s', 'speed': '92.4x'}
{'frame': '49375', 'fps': '2781', 'q': '-1.0', 'size': '301312kB', 'time': '00:27:27.41', 'bitrate': '1498.3kbits/s', 'speed': '92.8x'}
{'frame': '50825', 'fps': '2784', 'q': '-1.0', 'size': '307712kB', 'time': '00:28:15.83', 'bitrate': '1486.5kbits/s', 'speed': '92.9x'}
{'frame': '52374', 'fps': '2792', 'q': '-1.0', 'size': '314624kB', 'time': '00:29:07.51', 'bitrate': '1474.9kbits/s', 'speed': '93.2x'}
{'frame': '53875', 'fps': '2798', 'q': '-1.0', 'size': '321792kB', 'time': '00:29:57.59', 'bitrate': '1466.5kbits/s', 'speed': '93.3x'}
{'frame': '55374', 'fps': '2803', 'q': '-1.0', 'size': '329728kB', 'time': '00:30:47.61', 'bitrate': '1462.0kbits/s', 'speed': '93.5x'}
{'frame': '56990', 'fps': '2813', 'q': '-1.0', 'size': '338176kB', 'time': '00:31:41.53', 'bitrate': '1456.9kbits/s', 'speed': '93.9x'}
{'frame': '58597', 'fps': '2823', 'q': '-1.0', 'size': '348416kB', 'time': '00:32:35.13', 'bitrate': '1459.9kbits/s', 'speed': '94.2x'}
{'frame': '60171', 'fps': '2831', 'q': '-1.0', 'size': '356352kB', 'time': '00:33:27.67', 'bitrate': '1454.0kbits/s', 'speed': '94.4x'}
{'frame': '61729', 'fps': '2837', 'q': '-1.0', 'size': '365568kB', 'time': '00:34:19.63', 'bitrate': '1454.0kbits/s', 'speed': '94.7x'}
{'frame': '63292', 'fps': '2844', 'q': '-1.0', 'size': '376064kB', 'time': '00:35:11.78', 'bitrate': '1458.8kbits/s', 'speed': '94.9x'}
{'frame': '64898', 'fps': '2852', 'q': '-1.0', 'size': '384000kB', 'time': '00:36:05.37', 'bitrate': '1452.7kbits/s', 'speed': '95.1x'}
{'frame': '66480', 'fps': '2858', 'q': '-1.0', 'size': '393472kB', 'time': '00:36:58.15', 'bitrate': '1453.2kbits/s', 'speed': '95.4x'}
{'frame': '68007', 'fps': '2862', 'q': '-1.0', 'size': '400640kB', 'time': '00:37:49.11', 'bitrate': '1446.4kbits/s', 'speed': '95.5x'}
{'frame': '69542', 'fps': '2867', 'q': '-1.0', 'size': '408832kB', 'time': '00:38:40.32', 'bitrate': '1443.4kbits/s', 'speed': '95.7x'}
{'frame': '71141', 'fps': '2873', 'q': '-1.0', 'size': '417536kB', 'time': '00:39:33.67', 'bitrate': '1441.0kbits/s', 'speed': '95.9x'}
{'frame': '72751', 'fps': '2880', 'q': '-1.0', 'size': '426496kB', 'time': '00:40:27.39', 'bitrate': '1439.3kbits/s', 'speed': '96.1x'}
{'frame': '74265', 'fps': '2883', 'q': '-1.0', 'size': '434432kB', 'time': '00:41:17.94', 'bitrate': '1436.2kbits/s', 'speed': '96.2x'}
update_button_visible
mc.status.player_state UNKNOWN /home/chris/Downloads/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv -1130071249199169886
CastStatus(is_active_input=False, is_stand_by=True, volume_level=1.0, volume_muted=False, app_id='CC1AD845', display_name='Default Media Receiver', namespaces=['urn:x-cast:com.google.cast.cac', 'urn:x-cast:com.google.cast.debugoverlay', 'urn:x-cast:com.google.cast.broadcast', 'urn:x-cast:com.google.cast.media'], session_id='63815ada-d9d4-42d2-b8b7-c650d02854db', transport_id='63815ada-d9d4-42d2-b8b7-c650d02854db', status_text='Default Media Receiver')
<MediaStatus {'metadata_type': None, 'title': None, 'series_title': None, 'season': None, 'episode': None, 'artist': None, 'album_name': None, 'album_artist': None, 'track': None, 'subtitle_tracks': {}, 'images': [], 'supports_pause': False, 'supports_seek': False, 'supports_stream_volume': False, 'supports_stream_mute': False, 'supports_skip_forward': False, 'supports_skip_backward': False, 'current_time': 0, 'content_id': None, 'content_type': None, 'duration': None, 'stream_type': 'UNKNOWN', 'idle_reason': None, 'media_session_id': None, 'playback_rate': 1, 'player_state': 'UNKNOWN', 'supported_media_commands': 0, 'volume_level': 1, 'volume_muted': False, 'media_custom_data': {}, 'media_metadata': {}, 'current_subtitle_tracks': [], 'last_updated': None}>
[('Content-Type', ''), ('Content-Length', '0'), ('Host', '192.168.1.72:38363'), ('Connection', 'keep-alive'), ('User-Agent', 'Mozilla/5.0 (X11; Linux armv7l) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.81 Safari/537.36 CrKey/1.42.172094'), ('Accept-Encoding', 'identity;q=1, *;q=0'), ('Accept', '*/*'), ('Cast-Device-Capabilities', '{"bluetooth_supported":true,"display_supported":true,"hi_res_audio_supported":false,"remote_control_input_supported":false,"touch_input_supported":false}'), ('Accept-Language', 'en-us,en'), ('Range', 'bytes=0-')]
ranges [(0, 1000000000000)]
192.168.1.70 - - [08/Oct/2019:12:32:21 -0500] "GET /media/-1130071249199169886.mkv HTTP/1.1" 206 451407050 "-" "Mozilla/5.0 (X11; Linux armv7l) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.81 Safari/537.36 CrKey/1.42.172094"
[('Content-Type', ''), ('Content-Length', '0'), ('Host', '192.168.1.72:38363'), ('Connection', 'keep-alive'), ('User-Agent', 'Mozilla/5.0 (X11; Linux armv7l) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.81 Safari/537.36 CrKey/1.42.172094'), ('Accept-Encoding', 'identity;q=1, *;q=0'), ('Accept', '*/*'), ('Cast-Device-Capabilities', '{"bluetooth_supported":true,"display_supported":true,"hi_res_audio_supported":false,"remote_control_input_supported":false,"touch_input_supported":false}'), ('Accept-Language', 'en-us,en'), ('Range', 'bytes=449445888-')]
ranges [(449445888, 1000000000000)]
192.168.1.70 - - [08/Oct/2019:12:32:21 -0500] "GET /media/-1130071249199169886.mkv HTTP/1.1" 206 1961162 "-" "Mozilla/5.0 (X11; Linux armv7l) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.81 Safari/537.36 CrKey/1.42.172094"
update_button_visible
[('Content-Type', ''), ('Content-Length', '0'), ('Host', '192.168.1.72:38363'), ('Connection', 'keep-alive'), ('User-Agent', 'Mozilla/5.0 (X11; Linux armv7l) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.81 Safari/537.36 CrKey/1.42.172094'), ('Accept-Encoding', 'identity;q=1, *;q=0'), ('Accept', '*/*'), ('Cast-Device-Capabilities', '{"bluetooth_supported":true,"display_supported":true,"hi_res_audio_supported":false,"remote_control_input_supported":false,"touch_input_supported":false}'), ('Accept-Language', 'en-us,en'), ('Range', 'bytes=32768-')]
ranges [(32768, 1000000000000)]
192.168.1.70 - - [08/Oct/2019:12:32:23 -0500] "GET /media/-1130071249199169886.mkv HTTP/1.1" 206 451374282 "-" "Mozilla/5.0 (X11; Linux armv7l) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.81 Safari/537.36 CrKey/1.42.172094"
disabled screensaver
update_button_visible
restored screensaver
update_button_visible

**Comment by ckrzen** [comment]
SCREENSHOT:


![Screenshot from 2019-10-08 12-32-46](https://user-images.githubusercontent.com/6942277/66418555-f46c1b00-e9c7-11e9-9121-faeac549ad1d.png)

**Comment by ckrzen** [comment]
Re-downloaded from source as: "mp4  1920x1080  1080p 2694k , avc1.640028, 30fps, video only, 353.08MiB" and caused the transcoder to crash again:


$ gnomecast 
update_button_visible
None
update_button_visible
update_button_visible
serving on http://192.168.1.72:58737
check_ffmpeg
b'/usr/bin/ffmpeg\n'
Chromecast('192.168.1.70', port=8009, device=DeviceStatus(friendly_name='Poodle Lounge TV', model_name='Chromecast', manufacturer='Unknown manufacturer', uuid=UUID('29466b4d-975e-ea46-7fdd-9c91dc866bff'), cast_type='cast'))
update_button_visible
update_button_visible
update_button_visible
Open clicked
File selected: ['/home/chris/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mp4']
update_button_visible
STOP command requested but no session is active.
STOP command requested but no session is active.
update_button_visible
update_button_visible
update_button_visible
video: Stream #0:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p(tv, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 0 kb/s, 29.97 fps, 29.97 tbr, 30k tbn, 59.94 tbc (default) '0:0'
FileMetadata(fn:/home/chris/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mp4, ready:True, thumbnail_fn:/tmp/gnomecast_thumbnail_069joz_v.jpg, container:mp4, video_streams:[StreamMetadata(index:0:0, codec:h264, title:und)], audio_streams:[], subtitles:[])
Exception in thread Thread-10:
Traceback (most recent call last):
  File "/usr/lib64/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/usr/lib64/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/home/chris/.local/lib/python3.7/site-packages/gnomecast.py", line 1092, in update_transcoders
    if not self.audio_stream: self.audio_stream = fmd.audio_streams[0]
IndexError: list index out of range

chose subtitle No subtitles. None None
STOP command requested but no session is active.




![Screenshot from 2019-10-08 12-45-58](https://user-images.githubusercontent.com/6942277/66419662-15ce0680-e9ca-11e9-8112-1564cdc348c7.png)

**Comment by keredson** [comment]
can you post the output of `ffprobe -i <that file>`?  or a download location?

re the no sound: how long did you let it play?  i recently switched it to ac3 for 5.1 surround sound support, but on my receiver occasionally the sound takes a bit.  (maybe up to 20s or so)

**Comment by ckrzen** [comment]
```
$ ffprobe -i Downloads/Joel\ Edgerton\ to\ Return\ as\ Owen\ Lars\ in\ Obi-Wan\ Series\ -\ Jedi\ Council-pCU2GE07KW4.mkv 
ffprobe version 4.1.4 Copyright (c) 2007-2019 the FFmpeg developers
  built with gcc 9 (GCC)
  configuration: --prefix=/usr --bindir=/usr/bin --datadir=/usr/share/ffmpeg --docdir=/usr/share/doc/ffmpeg --incdir=/usr/include/ffmpeg --libdir=/usr/lib64 --mandir=/usr/share/man --arch=x86_64 --optflags='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection' --extra-ldflags='-Wl,-z,relro -Wl,--as-needed -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld ' --extra-cflags=' ' --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-version3 --enable-bzlib --disable-crystalhd --enable-fontconfig --enable-frei0r --enable-gcrypt --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libcdio --enable-libdrm --enable-libjack --enable-libfreetype --enable-libfribidi --enable-libgsm --enable-libmp3lame --enable-nvenc --enable-openal --enable-opencl --enable-opengl --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-librsvg --enable-libsoxr --enable-libspeex --enable-libssh --enable-libtheora --enable-libvorbis --enable-libv4l2 --enable-libvidstab --enable-libvmaf --enable-libvpx --enable-libx264 --enable-libx265 --enable-libxvid --enable-libzvbi --enable-avfilter --enable-avresample --enable-postproc --enable-pthreads --disable-static --enable-shared --enable-gpl --disable-debug --disable-stripping --shlibdir=/usr/lib64 --enable-libmfx --enable-runtime-cpudetect
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
  libpostproc    55.  3.100 / 55.  3.100
Input #0, matroska,webm, from 'Downloads/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv':
  Metadata:
    COMPATIBLE_BRANDS: iso6avc1mp41
    MAJOR_BRAND     : dash
    MINOR_VERSION   : 0
    ENCODER         : Lavf58.20.100
  Duration: 00:41:45.28, start: -0.007000, bitrate: 1303 kb/s
    Stream #0:0: Video: h264 (High), yuv420p(tv, bt709, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 29.97 fps, 29.97 tbr, 1k tbn, 59.94 tbc (default)
    Metadata:
      HANDLER_NAME    : ISO Media file produced by Google Inc.
      DURATION        : 00:41:45.269000000
    Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
    Metadata:
      DURATION        : 00:41:45.281000000
```

**Comment by ckrzen** [comment]
Just started it again and we're past the "1m 36s" mark and no sound. 

Download command: `$ youtube-dl "https://youtu.be/pCU2GE07KW4"`

**Comment by ckrzen** [comment]
Output of the mpv command:

```
$ mpv Downloads/Joel\ Edgerton\ to\ Return\ as\ Owen\ Lars\ in\ Obi-Wan\ Series\ -\ Jedi\ Council-pCU2GE07KW4.mkv
Playing: Downloads/Joel Edgerton to Return as Owen Lars in Obi-Wan Series - Jedi Council-pCU2GE07KW4.mkv
 (+) Video --vid=1 (*) (h264 1920x1080 29.970fps)
 (+) Audio --aid=1 --alang=eng (*) (opus 2ch 48000Hz)
AO: [pulse] 48000Hz stereo 2ch float
VO: [gpu] 1920x1080 yuv420p
AV: 00:03:08 / 00:41:45 (7%) A-V:  0.000


Exiting... (Quit)
```

**Comment by ckrzen** [comment]
Combined install/update command for `gnomecast` and `youtube-dl`:

`$ pip3 install --upgrade --user youtube_dl gnomecast`

**Comment by ckrzen** [comment]
BTW: _**THANKS**_ for `gnomecast` ... it's awesome!

**Comment by ckrzen** [comment]
```
$ neofetch 
          /:-------------:\          chris@CKR-4 
       :-------------------::        ----------- 
     :-----------/shhOHbmp---:\      OS: Fedora release 30 (Thirty) x86_64 
   /-----------omMMMNNNMMD  ---:     Host: 20AMS53B00 ThinkPad X240 
  :-----------sMMMMNMNMP.    ---:    Kernel: 5.2.17-200.fc30.x86_64 
 :-----------:MMMdP-------    ---\   Uptime: 19 hours, 43 mins 
,------------:MMMd--------    ---:   Packages: 1781 (rpm) 
:------------:MMMd-------    .---:   Shell: bash 5.0.7 
:----    oNMMMMMMMMMNho     .----:   Resolution: 1366x768 
:--     .+shhhMMMmhhy++   .------/   DE: GNOME 3.32.2 
:-    -------:MMMd--------------:    WM: Mutter 
:-   --------/MMMd-------------;     WM Theme: Adwaita 
:-    ------/hMMMy------------:      Theme: Adwaita-dark [GTK2/3] 
:-- :dMNdhhdNMMNo------------;       Icons: Adwaita [GTK2/3] 
:---:sdNMMMMNds:------------:        Terminal: gnome-terminal 
:------:://:-------------::          CPU: Intel i5-4300U (4) @ 2.900GHz 
:---------------------://            GPU: Intel Haswell-ULT 
                                     Memory: 1810MiB / 3820MiB 

```

**Comment by keredson** [comment]
you're welcome!  glad you like!

ok, so the regular chromecast can't do ac3.  i'm going to have to add per-device config now.

for an immediate patch, if you change `ac3` on line 276 to `mp3` you should get sound.

**Comment by ckrzen** [comment]
You da man!

Oct 8, 2019, 17:43 by notifications@github.com:

**Comment by keredson** [comment]
fixed in v1.9.4.  should auto-select now to mp3 for non-ultra chromecasts and anything that isn't 5.1 or 7.1.

unknown devices will assume ac3 support.  curious to see what comes of that.  :)
